src/result.jl

```diff
@@ -0,0 +1,41 @@
+const LINT_RESULT_COLORS = Dict(
+    :INFO => :bold,
+    :WARN => :yellow,
+    :ERROR => :magenta,
+    :OK => :green)
+
+"""
+A collection of `LintMessage`s. This behaves similarly to a vector of
+`LintMessage`s, but has different display behaviour.
+"""
+immutable LintResult <: AbstractVector{LintMessage}
+    messages::Array{LintMessage, 1}
+end
+
+function Base.show(io::IO, res::LintResult)
+    print(io, "LintResult(")
+    show(io, res.messages)
+    print(io, ")")
+end
+@compat function Base.show(io::IO, ::MIME"text/plain", res::LintResult)
+    if isempty(res)
+        Base.print_with_color(LINT_RESULT_COLORS[:OK], io, "No issues found.")
```

Comment:
After playing with it a bit. I think it looks better when success returns nothing.